File: codility/lessons/5 - prefix sums/passing cars/passing_cars.js
// https://app.codility.com/programmers/lessons/5-prefix_sums/passing_cars/

function solution(array) {
    let passingCarsPairs = 0;

    for (let i = 0, east = 0, west, passingCarsLimit = 1000000000; i < array.length; i++)
        if ((west = array[i])) {
            if ((passingCarsPairs += east) > passingCarsLimit)
                return -1;
        } else
            east++;

    return passingCarsPairs;
}

File: codility/lessons/5 - prefix sums/passing cars/passing_cars.py
# https://app.codility.com/programmers/lessons/5-prefix_sums/passing_cars/

def solution(array):
    passing_cars_pairs = 0
    i, east, passing_cars_limit = 0, 0, 1_000_000_000

    for west in array:
        if west:
            passing_cars_pairs += east
            if passing_cars_pairs > passing_cars_limit:
                return -1
        else:
            east += 1

    return passing_cars_pairs
